src/network/connection.cpp

```diff
@@ -2102,17 +2109,16 @@ void ConnectionReceiveThread::receive()
 
 	unsigned int loop_count = 0;
 
-	/* first of all read packets from socket */
-	/* check for incoming data available */
-	while( (loop_count < 10) &&
-			(m_connection->m_udpSocket.WaitData(50))) {
+	// First read packets from socket
+	// Check for incoming data available
+	while ((loop_count < 10) && wait(100 * 1000, readset)) {
```

Comment:
maybe we could use a g_setting instead of a static 100*1000 (and cache it). And also 100 sec timeout is too large. Please set 30sec instead, like a classic TCP connection